DataFormats/L1TrackTrigger/interface/TTTrack.h

```diff
@@ -156,34 +189,92 @@ TTTrack<T>::TTTrack(std::vector<edm::Ref<edmNew::DetSetVector<TTStub<T> >, TTStu
   theMomentum = GlobalVector(0.0, 0.0, 0.0);
   theRInv = 0.0;
   thePOCA = GlobalPoint(0.0, 0.0, 0.0);
+  theD0 = 0.;
+  theZ0 = 0.;
+  theTanL = 0;
+  thePhi = 0;
+  theTrkMVA1 = 0;
+  theTrkMVA2 = 0;
+  theTrkMVA3 = 0;
   thePhiSector = 0;
-  theTrackSeed = 0;
+  theEtaSector = 0;
+  theTrackSeedType = 0;
   theChi2 = 0.0;
   theStubPtConsistency = 0.0;
-  NumFitPars = 0;
+  numFitPars = 0;
 }
 
 /// Meant to be default constructor
 template <typename T>
 TTTrack<T>::TTTrack(double aRinv,
-                    double aphi,
-                    double aeta,
+                    double aphi0,
+                    double aTanlambda,
                     double az0,
                     double ad0,
                     double aChi2,
+                    double trkMVA1,
+                    double trkMVA2,
+                    double trkMVA3,
                     unsigned int aHitPattern,
                     unsigned int nPar,
                     double aBfield) {
   theStubRefs.clear();
-  double thePT = MagConstant * aRinv * aBfield;
-  theMomentum = GlobalVector(GlobalVector::Cylindrical(thePT, aphi, thePT * sinh(aeta)));
+  double thePT = fabs(MagConstant * 1.0 / aRinv * aBfield / 100.0);  // Rinv is in cm-1
+  theMomentum = GlobalVector(GlobalVector::Cylindrical(thePT, aphi0, thePT * aTanlambda));
   theRInv = aRinv;
-  thePOCA = GlobalPoint(ad0 * cos(aphi), ad0 * sin(aphi), az0);
-  thePhiSector = 0;  // must be set externally
-  theTrackSeed = 0;  // must be set externally
+  thePOCA = GlobalPoint(ad0 * cos(aphi0), ad0 * sin(aphi0), az0);
+  theD0 = ad0;
+  theZ0 = az0;
+  thePhi = aphi0;
+  theTanL = aTanlambda;
+  thePhiSector = 0;      // must be set externally
+  theEtaSector = 0;      // must be set externally
+  theTrackSeedType = 0;  // must be set externally
   theChi2 = aChi2;
+  theTrkMVA1 = trkMVA1;
+  theTrkMVA2 = trkMVA2;
+  theTrkMVA3 = trkMVA3;
   theStubPtConsistency = 0.0;  // must be set externally
-  NumFitPars = nPar;
+  numFitPars = nPar;
+  theHitPattern = aHitPattern;
+  theBField = aBfield;
+  // should probably fill the momentum vectur
+}
+
+/// Second default constructor with split chi2
```

Comment:
rather than duplicating all this code, call the previous constructor and then just change the chi2 values?

Reply:
done